### PR TITLE
feat: Add support for Accept-Encoding [DEV-2205]

### PR DESCRIPTION
**Pull request by anikitinDSR**

### CODE REVIEW

**Comment by ankurdotb** [comment]
Task linked: [DEV-2205 Implement HTTP content encoding/compression based on request headers](https://app.clickup.com/t/6600954/DEV-2205)

File: go.mod [hunk 1]
```diff
@@ -31,7 +43,7 @@ require (
 	github.com/go-openapi/swag v0.22.3 // indirect
 	github.com/gogo/protobuf v1.3.3 // indirect
 	github.com/golang-jwt/jwt v3.2.2+incompatible // indirect
-	github.com/golang/protobuf v1.5.2 // indirect
+	github.com/golang/protobuf v1.5.3 // indirect
```
Comment (abdulla-ashurov) [inline go.mod:46]: Is it OK? We changed protobuf version in go.mod file.

File: tests/unit-tests/serve_test.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	"net/http/httptest"
+
+	diddocServices "github.com/cheqd/did-resolver/services/diddoc"
+	"github.com/cheqd/did-resolver/types"
+	"github.com/labstack/echo/v4"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Content/Accept encoding checks", func() {
```
Comment (abdulla-ashurov) [inline tests/unit-tests/serve_test.go:13]: Do we need to add unit tests for testing any accept-encoding.

Comment (anikitinDSR) [reply tests/unit-tests/serve_test.go:14]: Hey! Yes, we've already added unit tests. Thanks for the suggestion

File: utils/print_config.go [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package utils
```
Comment (ankurdotb) [inline utils/print_config.go:1]: I'd probably move `serve.go` to this one, and just make it `config.go` where configurations can be set/printed. What do you think?

File: utils/serve.go [hunk 4]
```diff
@@ -0,0 +1,26 @@
+package utils
+
+import (
+	"github.com/cheqd/did-resolver/types"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+)
```
Comment (ankurdotb) [inline utils/serve.go:7]: Is this necessary any more? See comment on above (since 'serve' is happening in main now)

File: main.go [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+//	@contact.name	Cheqd Foundation Limited
+//	@contact.url	https://cheqd.io
+//	@license.name	Apache 2.0
+//	@license.url	https://github.com/cheqd/did-resolver/blob/main/LICENSE
+//	@host			resolver.cheqd.net
+//	@BasePath		/1.0/identifiers
+//	@schemes		https http
+
+func main() {
+	utils.PrintConfig()
```
Comment (ankurdotb) [inline main.go:89]: Fails Golang linter on this function call. See output:

```bash
ankur@Ankurs-MacBook-Pro ~/w/l/did-resolver (DEV-2205) [SIGINT]> golangci-lint run -c .github/linters/.golangci.yaml --fix
main.go:89:19: Error return value of `utils.PrintConfig` is not checked (errcheck)
        utils.PrintConfig()
                         ^
```

**Review by ankurdotb — CHANGES_REQUESTED** [review]
One minor linter issue to be fixed

**Pull request merged**